source/ncengine/graphics2/diligent/pass/Pass.cpp

```diff
@@ -125,12 +132,14 @@ cbuffer EnvironmentBufferData
 
 void main(in  VSInput VSIn, uint InstanceID : SV_InstanceID,  out PSInput PSIn)
 {
-    float4 TransformedPos = mul(float4(VSIn.Pos, 1.0), MeshRendererBufferData[InstanceID].model);
+    uint transformIndex = InstanceBufferData[InstanceID].transformIndex;
+    uint materialIndex = InstanceBufferData[InstanceID].materialIndex;
+    float4 TransformedPos = mul(float4(VSIn.Pos, 1.0), TransformBufferData[transformIndex].model);
     PSIn.Pos = mul(TransformedPos, cameraViewProjection);
     PSIn.UV  = VSIn.UV;
-    PSIn.Normal = normalize(mul(MeshRendererBufferData[InstanceID].model, VSIn.Normal)); // @TODO #805, compute inverse model matrix CPU-side
+    PSIn.Normal = normalize(mul(TransformBufferData[transformIndex].model, VSIn.Normal)); // @TODO #805, compute inverse model matrix CPU-side
```

Comment:
With this new architecture - should extra 'Per Instance' data live in `TransformData` now? (In this case it is a transform that would need to be added: the inverse transpose of the model matrix) I may shift to using the TBN matrix and doing lighting in tangent space anyway- so if it throws a wrench in the gears I can go that route.

Reply:
Generally, its a toss up between `TransformData` and `InstanceData`. As discussed, we'll probably end up with variants of `InstanceData` (mesh vs. skinned renderer, etc) with slightly different properties, so it probably depends on whether it pertains to all instance types or a subset. For an additional matrix, I'd assume `TransformData`.